### DISCUSSION TITLE
Riing Trio Controller shows up as Quad and is not recognized after reboot

### DISCUSSION THREAD

**Discussion by benjj-wr** [devcompl/TTController#106] [General]
First of all, thank you so much for this program. I love the look of the Thermaltake Fans but their software is beyond garbage. This program is exactly what I was looking for and I'm so glad I found this repo. 

So I have 6 Riing Trio Fans installed, connected via two controllers (5 fans controller 1; 1 fan controller 2).
After I set everything up and configured the config.json, everything except the one fan on controller 2 worked, but oddly the second controller is recognized as a Riing Quad controller. Just addressed those ports in the config as well, worked fine after restarting the service. But after a reboot of my PC, the one controller was set back to default and the service didn't recognize the second controller anymore. After unplugging the controller and plugging it back in, it worked again, but still recognized as Riing Quad controller.
Is this just a faulty controller or did TT just ship me the wrong product? Maybe my config is wrong somewhere? Did anyone else have a similar problem? 
Here is the full output of the debug report:

```
Info
-------------------------------
»
»        OS: Windows 10 Home 1909 [18362.1.wow64fre.19h1_release.190318-1202]
»        Build: 0.11.0-beta+Branch.master.Sha.42768ba987b22e18ff8cdd114f754e5f7ce16d72
»
-------------------------------
Applications
-------------------------------
»
»
-------------------------------
HID
-------------------------------
»
»  [0x264a, 0x2260]: \\?\hid#vid_264a&pid_2260#8&38997234&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
»  [0x264a, 0x2135]: \\?\hid#vid_264a&pid_2135#8&149dc185&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
»
-------------------------------
Controllers
-------------------------------
»
»  Name: Riing Quad
»        VendorId: 9802
»        ProductId: 8800
»        Version: 3.0.3
»        Ports:
»                1:
»                        Data: [PortId: 1, Speed: 35, Rpm: 512, Unknown: 255]
»                        Identifier: [9802, 8800, 1]
»                2:
»                        Data: [PortId: 2, Speed: 35, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8800, 2]
»                3:
»                        Data: [PortId: 3, Speed: 35, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8800, 3]
»                4:
»                        Data: [PortId: 4, Speed: 35, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8800, 4]
»                5:
»                        Data: [PortId: 5, Speed: 35, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8800, 5]
»        Available effect types: PerLed
»  Name: Riing Trio
»        VendorId: 9802
»        ProductId: 8501
»        Version: 3.0.1
»        Ports:
»                1:
»                        Data: [PortId: 1, Speed: 35, Rpm: 404, Unknown: 255]
»                        Identifier: [9802, 8501, 1]
»                2:
»                        Data: [PortId: 2, Speed: 35, Rpm: 460, Unknown: 255]
»                        Identifier: [9802, 8501, 2]
»                3:
»                        Data: [PortId: 3, Speed: 35, Rpm: 420, Unknown: 255]
»                        Identifier: [9802, 8501, 3]
»                4:
»                        Data: [PortId: 4, Speed: 35, Rpm: 314, Unknown: 255]
»                        Identifier: [9802, 8501, 4]
»                5:
»                        Data: [PortId: 5, Speed: 35, Rpm: 391, Unknown: 255]
»                        Identifier: [9802, 8501, 5]
»        Available effect types: PerLed
»
-------------------------------
Sensors
-------------------------------
»
»  Nuvoton NCT6797D:
»        Temperature:
»                CPU Core (/lpc/nct6797d/temperature/0): 48,5 °C
»                Temperature #1 (/lpc/nct6797d/temperature/1): 46,5 °C
»                Temperature #2 (/lpc/nct6797d/temperature/2): 36,0 °C
»                Temperature #3 (/lpc/nct6797d/temperature/3): 36,0 °C
»                Temperature #4 (/lpc/nct6797d/temperature/4): 33,0 °C
»                Temperature #5 (/lpc/nct6797d/temperature/5): 39,0 °C
»                Temperature #6 (/lpc/nct6797d/temperature/6): -1,0 °C
»
»  Intel Core i7-9700K:
»        Temperature:
»                CPU Core #1 (/intelcpu/0/temperature/0): 43,0 °C
»                CPU Core #2 (/intelcpu/0/temperature/1): 47,0 °C
»                CPU Core #3 (/intelcpu/0/temperature/2): 41,0 °C
»                CPU Core #4 (/intelcpu/0/temperature/3): 43,0 °C
»                CPU Core #5 (/intelcpu/0/temperature/4): 42,0 °C
»                CPU Core #6 (/intelcpu/0/temperature/5): 42,0 °C
»                CPU Core #7 (/intelcpu/0/temperature/6): 41,0 °C
»                CPU Core #8 (/intelcpu/0/temperature/7): 40,0 °C
»                CPU Package (/intelcpu/0/temperature/8): 47,0 °C
»                CPU Core #1 Distance to TjMax (/intelcpu/0/temperature/9): 57,0 °C
»                CPU Core #2 Distance to TjMax (/intelcpu/0/temperature/10): 53,0 °C
»                CPU Core #3 Distance to TjMax (/intelcpu/0/temperature/11): 59,0 °C
»                CPU Core #4 Distance to TjMax (/intelcpu/0/temperature/12): 57,0 °C
»                CPU Core #5 Distance to TjMax (/intelcpu/0/temperature/13): 58,0 °C
»                CPU Core #6 Distance to TjMax (/intelcpu/0/temperature/14): 58,0 °C
»                CPU Core #7 Distance to TjMax (/intelcpu/0/temperature/15): 59,0 °C
»                CPU Core #8 Distance to TjMax (/intelcpu/0/temperature/16): 60,0 °C
»                Core Max (/intelcpu/0/temperature/17): 47,0 °C
»                Core Average (/intelcpu/0/temperature/18): 42,4 °C
»
»  NVIDIA GeForce RTX 2080 Ti:
»        Temperature:
»                GPU Core (/gpu-nvidia/0/temperature/0): 36,0 °C
»
»  TOSHIBA DT01ACA300:
»        Temperature:
»                Temperature (/hdd/1/temperature/0): 35,0 °C
»
»  Samsung SSD 970 EVO Plus 1TB:
»        Temperature:
»                Temperature (/nvme/0/temperature/0): 51,0 °C
»                Temperature 1 (/nvme/0/temperature/6): 51,0 °C
»                Temperature 2 (/nvme/0/temperature/7): 69,0 °C
»
»  WDC WD10EADS-00L5B1:
»        Temperature:
»                Temperature (/hdd/2/temperature/0): 32,0 °C
»
-------------------------------
Press any key to continue...
```

**Comment by MoshiMoshi0** [comment]
> Is this just a faulty controller or did TT just ship me the wrong product? 

I think thermaltake started to use riing quad controllers in every fan package. Its easier to handle only one controller than to have a different controller for each fan.

> Maybe my config is wrong somewhere? Did anyone else have a similar problem?

Similar issue would be #103. In general the issue is well known, there were a few people here that reported not detected controllers, and on the official forum you can see a lot of threads about it.

If its not visible in HID section than this is not a config or service issue. I think there must be some hardware problem in the controller, and/or it also might be linked to the motherboard.
You can try searching for devices that have `idVendor = 0x264A` in [USBView](https://www.ftdichip.com/Support/Utilities.htm#MicrosoftUSBView), if you can find 2 even if 1 is detected in the service, please post the text on the right side.
If you can find only 1, then even windows cant find it.

Then i would try connecting each controller to a different usb header, you can also try using simple usb -> micro usb cable and connect them to the front/back usb ports to test. You can also test a different pc.
If nothing helps and the controller is still periodically not working then I think you would need to RMA. I dont think the cause of this issue is known, intermittent problems are the worst.

**Reply by benjj-wr** [reply]
Hey Moshi, thanks so much for your input. I have tried pretty much everything at this point, every cable configuriation, only use the controller in question etc. Yet it still does not show up after a reboot, only when I replug the Molex Power connector of the controller it is recognized. I suppose this is just a faulty controller and I will probably proceed to RMA or find a different solution. Thank you so much tho!

**Comment by benjj-wr** [comment]
Okay, I discovered something new. Through all my testing I did not have the TT RGB Plus software installed obviously. I just reinstalled out of curiosity and suddenly the controller gets recognized when I reboot. When I completely uninstall TT RGB Plus the controller somehow does not show up anymore. I honestly have no idea whats going on.

**Reply by MoshiMoshi0** [reply]
Thats interesting. Does it just have to be installed or it has to be running? Or maybe you have to run it once for the controller to be visible?
Did you check USBView before installing TT RGB Plus? Was it visible there or no?

**Reply by benjj-wr** [reply]
Hey. so that was a false positive. I now tested every scenario and it just seems to be random when it works and when not. When I replug it even without TT RGB Plus, it still shows up, but not after a reboot. Even with TT RGB Plus installed it sometimes does show up and sometimes doesn't. 
When it does not show up the controller is neither visible in USBView nor Device Manager. Only when I replug the power connection to the controller it shows up. I suspected it to be a power issue because I use a Molex Splitter, but I tested it and they give the same voltage output. I really think the controller is just defective. I even took everything out, plugged in only the faulty controller with TT RGB Plus installed and still get the famous "Controller not found Error: Hx00001" or sth.

**Reply by MoshiMoshi0** [reply]
Yea I dunno, it might be defective but it also might be a design flaw. Might just be a timing issue with power/usb initialization order, thats why I suggested a different pc.

Unfortunately I cant do much, if you find anything interesting let us know here.

**Reply by benjj-wr** [reply]
So just to conclude this thread, it was a defective controller after all. RMA'd both of them and got 2 new ones, both are instantly recognized on every boot cycle and never fail to connect. Still, thank you very much for your input!
